### PR TITLE
Issue#127 Implementation of notification

### LINKED ISSUE #127: Implementation of notification
### Description

As admin of application I want to invite headman of cooperation in application, so that he could be in charge of his cooperation.

### Requirements:

- send an invitation to user that we want to make headman of cooperation
- set role and cooperation to this user
- configure template for different types of emails
- send automatic notification at certain intervals to finish registration

### Acceptance
- [x] sending email with an invitation to system
- [ ] setting role and cooperation to headman of cooperation to finish registration

### PR DESCRIPTION
**Pull request by PoLytvy**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @alexorlenko 
- [ ] @kh0ma 
- [ ] @Romanov-Niko 
- [ ] @Abhai2016 
- [ ] @annaprok 
- [ ] @ElgorTheWanderer 

### Second Level Review

- [ ] @kh0ma 

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-data-migration/src/main/resources/db/changelog/0.0.1/Issue127-QuartzConfig.xml [hunk 1]
```diff
@@ -0,0 +1,181 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="Issue127.001" author="Dandria1802">
```
Comment (kh0ma) [inline home-data-migration/src/main/resources/db/changelog/0.0.1/Issue127-QuartzConfig.xml]: Can we reuse the one that we can take from classpath? e.g. add it into the master changelog:
`    <include file="classpath:org/quartz/impl/jdbcjobstore/liquibase.quartz.init.xml"/>`

File: home-application/pom.xml [hunk 2]
```diff
@@ -63,6 +63,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-actuator</artifactId>
         </dependency>
+<!--        <dependency>-->
```
Comment (Abhai2016) [inline home-application/pom.xml]: Interesting why we commented it...

File: home-data-migration/src/main/resources/db/changelog/0.0.1/Issue127-QuartzConfig.xml (hunk 1 above)
Comment (likeRewca) [reply home-data-migration/src/main/resources/db/changelog/0.0.1/Issue127-QuartzConfig.xml]: done

**Review by veniamin-aveltsev — APPROVED** [review]

File: home-service/src/main/resources/service.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#Path to text
+path.invitation.registration="template/invitation-to-registration.mustache"
+path.invitation.cooperation="template/invitation-to-cooperation.mustache"
+path.invitation.apartment="template/invitation-to-apartment.mustache"
+
+quartz.time.job=7200000
```
Comment (IRoss-git) [inline home-service/src/main/resources/service.properties]: add new line

File: home-service/src/test/java/com/softserveinc/ita/homeproject/homeservice/service/impl/MailServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    void sendTextMessageTest() throws MessagingException {
+        when(mailSender.createMimeMessage()).thenReturn(mimeMessage);
+        when(templateService.createMessageTextFromTemplate(mailDto)).thenReturn("Welcome to the club body!");
+
+        mailService.sendTextMessage(mailDto);
+        assertEquals("Welcome to the club body!", templateService.createMessageTextFromTemplate(mailDto));
+    }
+}
```
Comment (IRoss-git) [inline home-service/src/test/java/com/softserveinc/ita/homeproject/homeservice/service/impl/MailServiceImplTest.java]: Add new line as well

File: home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml [hunk 5]
```diff
@@ -12,4 +12,6 @@
     <include file="Issue91.xml" relativeToChangelogFile="true"/>
     <include file="Issue156.xml" relativeToChangelogFile="true"/>
     <include file="Issue180.xml" relativeToChangelogFile="true"/>
```
Comment (IRoss-git) [inline home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml:14]: Why did we add 2 times Issue180.xml?

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/quartz/jobs/SendCooperationEmailJob.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    private final MailService mailService;
+
+    public SendCooperationEmailJob(ServiceMapper mapper,
+                                   MailService mailService,
+                                   CooperationInvitationService invitationService,
+                                   UserRepository userRepository) {
+        this.mapper = mapper;
+        this.mailService = mailService;
+        this.invitationService = invitationService;
+        this.userRepository = userRepository;
```
Comment (IRoss-git) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/quartz/jobs/SendCooperationEmailJob.java]: Could we use lombok for constructor?

Comment (likeRewca) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/quartz/jobs/SendCooperationEmailJob.java]: Ok, if all written with @RequiredArgsConstructor I will do the same

File: home-application/pom.xml (hunk 2 above)
Comment (likeRewca) [reply home-application/pom.xml]: I don`t know why we need this on this module, and that is why I removed it

File: home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml (hunk 5 above)
Comment (likeRewca) [reply home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml:14]: done

File: home-service/src/main/resources/service.properties (hunk 3 above)
Comment (likeRewca) [reply home-service/src/main/resources/service.properties]: done

File: home-service/src/test/java/com/softserveinc/ita/homeproject/homeservice/service/impl/MailServiceImplTest.java (hunk 4 above)
Comment (likeRewca) [reply home-service/src/test/java/com/softserveinc/ita/homeproject/homeservice/service/impl/MailServiceImplTest.java]: done

File: home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml (hunk 5 above)
Comment (likeRewca) [reply home-data-migration/src/main/resources/db/changelog/0.0.1/_cumulative.xml:14]: done

**Review by IRoss-git — APPROVED** [review]

File: home-service/src/main/resources/application-home-service.properties [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+spring.mail.port=587
+spring.mail.username=${MAIL_USERNAME:test.ita.emails@gmail.com}
+spring.mail.password=${MAIL_PASSWORD:itacademy2021}
+spring.mail.properties.mail.smtp.auth=true
+spring.mail.properties.mail.smtp.starttls.enable=true
+
+#Quartz service
+spring.quartz.job-store-type = jdbc
+spring.quartz.properties.org.quartz.jobStore.class=org.quartz.impl.jdbcjobstore.JobStoreCMT
+spring.quartz.properties.org.quartz.jobStore.driverDelegateClass=org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
```
Comment (tex1988) [inline home-service/src/main/resources/application-home-service.properties:12]: I propose to rename this file to application.properties according to general naming of such files in the project

**Review by tex1988 — APPROVED** [review]

**Review by annaprok — APPROVED** [review]

**Review by annaprok — APPROVED** [review]

**Pull request merged**